projects/igniteui-angular/src/lib/select/select.component.spec.ts

```diff
@@ -491,18 +493,57 @@ describe('igxSelect', () => {
             const fix = TestBed.createComponent(IgxSelectReactiveFormComponent);
-
-            let inputGroupWithRequiredAsterisk = fix.debugElement.query(By.css('.' + CSS_CLASS_INPUT_GROUP_REQUIRED));
             fix.detectChanges();
+            // 1) check if label's --required class and its asterisk are applied
+            const dom = fix.debugElement;
             const selectComp = fix.componentInstance.select;
             const formGroup: FormGroup = fix.componentInstance.reactiveForm;
+            let inputGroupIsRequiredClass = dom.query(By.css('.' + CSS_CLASS_INPUT_GROUP_REQUIRED));
+            let inputGroupInvalidClass = dom.query(By.css('.' + CSS_CLASS_INPUT_GROUP_INVALID));
+            // interaction test - expect actual asterisk
+            // The only way to get a pseudo elements like :before OR :after is to use getComputedStyle(element [, pseudoElt]),
+            // as these are not in the actual DOM
+            let asterisk = window.getComputedStyle(dom.query(By.css('.' + CSS_CLASS_INPUT_GROUP_LABEL)).nativeElement, ':after').content;
+            expect(asterisk).toBe('"*"');
+            expect(inputGroupIsRequiredClass).toBeDefined();
+            expect(inputGroupIsRequiredClass).not.toBeNull();
+
+            // 2) check that input group's --invalid class is NOT applied
+            expect(inputGroupInvalidClass).toBeNull();
+
+            // interaction test - markAsTouched + open&close so the --invalid and --required classes are applied
+            fix.debugElement.componentInstance.markAsTouched();
+            const inputGroup = fix.debugElement.query(By.css('.' + CSS_CLASS_INPUT_GROUP));
+            inputGroup.nativeElement.click();
+            const toggleBtn = fix.debugElement.query(By.css('.' + CSS_CLASS_TOGGLE_BUTTON));
+            toggleBtn.nativeElement.click();
+            tick();
+            fix.detectChanges();
+            expect(selectComp.collapsed).toEqual(true);
+
+            inputGroupInvalidClass = dom.query(By.css('.' + CSS_CLASS_INPUT_GROUP_INVALID));
+            expect(inputGroupInvalidClass).not.toBeNull();
+            expect(inputGroupInvalidClass).not.toBeUndefined();
+
+            inputGroupIsRequiredClass = dom.query(By.css('.' + CSS_CLASS_INPUT_GROUP_REQUIRED));
+            expect(inputGroupIsRequiredClass).not.toBeNull();
+            expect(inputGroupIsRequiredClass).not.toBeUndefined();
+
+            // Check if the input group's --invalid and --required classes are removed when validator is dynamically cleared
             fix.componentInstance.removeValidators(formGroup);
+            fix.detectChanges();
+            tick();
+
+            inputGroupIsRequiredClass = dom.query(By.css('.' + CSS_CLASS_INPUT_GROUP_REQUIRED));
             const selectFormReference = fix.componentInstance.reactiveForm.controls.optionsSelect;
+            // interaction test - expect actual asterisk
```

Comment:
Shouldn't this be "expect MISSING asterisk"?

Reply:
done